### PR TITLE
Feature 3.7/parametrize truncate collect

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/12857

Enterprise companion: https://github.com/arangodb/enterprise/pull/565

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12336/

**Comment by dothebart** [comment]
tests blue.

**Review by jsteemann — APPROVED** [review]

File: js/client/modules/@arangodb/arango-collection.js [hunk 1]
```diff
@@ -533,22 +533,21 @@ ArangoCollection.prototype.truncate = function (options) {
...
   var requestResult = this._database._connection.PUT(this._baseurl('truncate') + append, null, headers);
-
   arangosh.checkRequestResult(requestResult);
   // invalidate cache
   this._status = null;
-
-  if (!options.compact) {
-    return;
-  }
+  return requestResult;
```
Comment (jsteemann) [inline js/client/modules/@arangodb/arango-collection.js:550]:
```suggestion
```

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12432/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**